### PR TITLE
Optimistic conversion as if single line

### PR DESCRIPTION
**Pull request by sungshik**
This PR:

  - Disables the check for single-line-ness to optimistically convert multi-line productions to single-line rules
  - To make the optimistic conversion work as expected, adds logic to sort productions based on their starting delimiter (if any)
  - To make the sorting work as expected, refactors and extends the logic to compute inner/outer delimiters of productions
  - Updates existing tests and adds new ones

### CODE REVIEW

File: rascal-textmate-core/src/main/rascal/lang/rascal/grammar/analyze/Delimiters.rsc [hunk 1]
```diff
@@ -10,101 +10,162 @@ import util::Maybe;
...
+                return just(s);
+            }
+            if (isNonTerminalType(s) && just(delimiter) := getInnerDelimitersBySymbol(g, direction)[s]) {
+                return just(delimiter);
+            }
+        }
+        return delimiters[delabel(def)];
+    }
+
+    solve (delimiters) delimiters = delimiters + (s: \do(s) | s <- todo());
```
Comment (DavyLandman) [inline rascal-textmate-core/src/main/rascal/lang/rascal/grammar/analyze/Delimiters.rsc:147]: also here, I think the code will be a bit more readable if not written as 4 nested functions?

**Review by DavyLandman — APPROVED** [review]
As far as I can understand, it looks good.

File: rascal-textmate-core/src/main/rascal/lang/rascal/grammar/analyze/Delimiters.rsc [hunk 2]
```diff
@@ -10,101 +10,162 @@ import util::Maybe;
...
 @synopsis{
-    Checks if a list contains a delimiter
+    Returns the single delimiter if set `delimiters` is a singleton. Returns
+    `nothing()` otherwise.
 }
 
-bool containsDelimiter(list[Symbol] symbols)
-    = any(s <- symbols, isDelimiter(s));
+Maybe[Symbol] unique(set[Maybe[Symbol]] delimiters)
+    = {d: just(_)} := delimiters
```
[the remark this answers is not in the document]
Comment (sungshik) [reply rascal-textmate-core/src/main/rascal/lang/rascal/grammar/analyze/Delimiters.rsc]: Fixed

**Review by sungshik — COMMENTED** [review]
Thank you Davy!

**Pull request merged**